docs/docs/Contributing/contributing-how-to-contribute.md

```diff
@@ -4,157 +4,115 @@ sidebar_position: 1
-6. Add the dependency to [/documentloaders/__init__.py](https://github.com/langflow-ai/langflow/blob/dev/src/backend/base/langflow/components/documentloaders/__init__.py) as `from .MyCustomDocumentLoader import MyCustomDocumentLoader`.
-7. Add any new dependencies to the outer [pyproject.toml](https://github.com/langflow-ai/langflow/blob/dev/pyproject.toml#L27) file.
-8. Submit documentation for your component. For this example, you'd submit documentation to the [loaders page](https://github.com/langflow-ai/langflow/blob/dev/docs/docs/components/loaders).
-9. Submit your changes as a pull request. The Langflow team will have a look, suggest changes, and add your component to Langflow.
-
-## User Sharing {#34ac32e11f344eab892b94531a21d2c9}
-
-
----
-
+2. Install dependencies and start the frontend:
 
-You might want to share and test your custom component with others, but don't need it merged into the main source code.
+```bash
+make frontend
+```
 
+## Contribute documentation
 
-If so, you can share your component on the Langflow store.
+The documentation is built using [Docusaurus](https://docusaurus.io/). To run the documentation locally, run the following commands:
 
+```bash
+cd docs
+npm install
+npm run start
+```
 
-1. [Register at the Langflow store](https://www.langflow.store/login/).
+The documentation will be available at `localhost:3000` and all the files are located in the `docs/docs` folder.
 
+## Open a pull request
 
-2. Undergo pre-validation before receiving an API key.
+Once you have written and manually tested your changes with `make lint` and `make unit_tests`, open a pull request to send your changes upstream to the main Langflow repository.
 
+1. Open a new GitHub pull request with your patch against the `main` branch.
+2. Ensure the PR title follows semantic commits conventions. For example, features are `feat: add new feature` and fixes are `fix: correct issue with X`.
+3. A Langflow maintainer will review your pull request. Thanks for your contribution!
 
-3. To deploy your amazing component directly to the Langflow store, without it being merged into the main source code, navigate to your flow, and then click **Share**. The share window appears:
+Some additional guidance on pull request titles:
+* Ensure the pull request description clearly describes the problem and solution. If the PR fixes an issue, include a link to the fixed issue in the PR description with `Fixes #1234`.
+* Pull request titles appear in Langflow's release notes, so they should explain what the PR does as explicitly as possible.
+* Pull requests should strive to fix one thing **only**, and should contain a good description of what is being fixed.
 
+For more information, see the [Python Developer's Guide](https://devguide.python.org/getting-started/pull-request-lifecycle/index.html#making-good-commits).
 
-![](./683296796.png)
+## Report bugs or suggest improvements
```

Comment:
Was is an oversight that this information is here and in a separate topic? I'm not sure why you have it in two places.